### PR TITLE
Add a 4.2 --> 4.3 migration guide

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

**Comment by derobins** [comment]
I'll fill in the functions and symbols in a future PR

File: doc/HDF-4.2-to-4.3-migration.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Most users should not have to make any changes to their applications, as long
+as they only use the public API as documented in the HDF4 Reference Manual.
+The single- and multi-file API calls are all unchanged and we have preserved
+the legacy HDF 3 API (found in `df.h`).
+
+If you HAVE used any of the undocumented functionality packeded with the library
+(e.g., `linklist.h`'s linked lists), you will either have to pull those
+source files out of an older version of the library for incorporation into
+your own code, or stick with HDF 4.2.16. Note that we don't plan to have
+additional HDF 4.2.x maintenance releases.
```
Comment (byrnHDF) [inline doc/HDF-4.2-to-4.3-migration.md:31]: Maybe a 4.2.17 with only build system changes (GH binaries) for those who don't want to move to 4.3?

File: doc/HDF-4.2-to-4.3-migration.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+* Most of `hbitio.h` has been moved to a private header file.
+* Most of `hfile.h` has been moved to a private header file.
+* `hdf.h` no longer pulls in the `hdfi.h` internal header file. See below for the implications of this.
+
+## Migration Steps
+
+Most users should not have to make any changes to their applications, as long
+as they only use the public API as documented in the HDF4 Reference Manual.
+The single- and multi-file API calls are all unchanged and we have preserved
+the legacy HDF 3 API (found in `df.h`).
```
[the remark this answers is not in the document]
Comment (derobins) [reply doc/HDF-4.2-to-4.3-migration.md:25]: Nope. HDF 3.2 or even earlier.

File: doc/HDF-4.2-to-4.3-migration.md (hunk 1 above)
Comment (derobins) [reply doc/HDF-4.2-to-4.3-migration.md:31]: I'm not going to commit to that in a migration guide

**Pull request merged**